frontend/demo/src/reducers/data/channels/index.ts

```diff
@@ -0,0 +1,26 @@
+import {ActionType, getType} from 'typesafe-actions';
+import {Channel} from '../../../model/Channel';
+import * as actions from '../../../actions/channel';
+import {unionWith} from 'lodash-es';
+
+type Action = ActionType<typeof actions>;
+
+export const initialState = [];
+
+const mergeChannels = (channels: Channel[], newChannels: Channel[]) =>
+  unionWith(newChannels, channels, (channelA: Channel, channelB: Channel) => {
+    return channelA.sourceChannelId === channelB.sourceChannelId;
```

Comment:
Just out of curiosity: Why the `sourceChannelId` and not the channel id? 🤔 We now ensure on the backend that source channel ids are unique

Reply:
Because the api returns the source channel id and not always the `id` and the `source_channel_id` is always there. See docs: https://docs.airy.co/api/http#channels .

If this is wrong I am happy to change it.

Reply:
The only one that does not return the id, is `channels.explore` as far as I can tell from the docs. So I would suggest we change this.

Reply:
Yes, should I open a PR for that?